From ac8ff4a2758d10bc03840deac41791ca5db92b78 Mon Sep 17 00:00:00 2001
From: Saranya Gopal <saranya.gopal@intel.com>
Date: Mon, 11 Aug 2014 22:20:26 +0530
Subject: [PATCH 1/1] EM/Thermal: Use compat_ioctl in acpi_thermal_rel driver

Move to compat_ioctl to use the acpi_thermal_rel ioctls
from 32 bit userspace as well.

Change-Id: Icb62456d95f9be618b459907826720a50159f860
Signed-off-by: Saranya Gopal <saranya.gopal@intel.com>
---
 drivers/acpi/acpi_thermal_rel.c  |    1 +
 include/linux/acpi_thermal_rel.h |    2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/acpi/acpi_thermal_rel.c b/drivers/acpi/acpi_thermal_rel.c
index ee94e46035c4..2d2004fd8d0e 100644
--- a/drivers/acpi/acpi_thermal_rel.c
+++ b/drivers/acpi/acpi_thermal_rel.c
@@ -371,6 +371,7 @@ static const struct file_operations acpi_thermal_rel_fops = {
 	.open		= acpi_thermal_rel_open,
 	.release	= acpi_thermal_rel_release,
 	.unlocked_ioctl	= acpi_thermal_rel_ioctl,
+	.compat_ioctl	= acpi_thermal_rel_ioctl,
 	.llseek		= no_llseek,
 };
 
diff --git a/include/linux/acpi_thermal_rel.h b/include/linux/acpi_thermal_rel.h
index f00700bc9d79..fc05bb3559bd 100644
--- a/include/linux/acpi_thermal_rel.h
+++ b/include/linux/acpi_thermal_rel.h
@@ -1,7 +1,7 @@
 #ifndef __ACPI_ACPI_THERMAL_H
 #define __ACPI_ACPI_THERMAL_H
 
-#include <asm/ioctl.h>
+#include <linux/compat.h>
 
 #define ACPI_THERMAL_MAGIC 's'
 
-- 
1.7.9.5

